File: apps/api/app/services/pricing.py
from __future__ import annotations

from typing import Literal

VehicleSize = Literal["small", "medium", "large"]

# Keep values mirrored with web pricing constants in apps/web/lib/pricing.ts.
SIZE_MULTIPLIERS: dict[VehicleSize, float] = {
    "small": 1.0,
    "medium": 1.15,
    "large": 1.30,
}


def get_size_multiplier(size: VehicleSize) -> float:
    """Returns one configured size multiplier."""
    return SIZE_MULTIPLIERS[size]


def get_adjusted_service_price(base_price: int | float, size: VehicleSize) -> int:
    """Returns a whole-dollar service price adjusted by vehicle size."""
    return round(float(base_price) * get_size_multiplier(size))

File: DetailFlow/apps/web/lib/pricing.ts
import type { VehicleSize } from '@/lib/booking-types';

export const SIZE_MULTIPLIERS: Record<VehicleSize, number> = {
  small: 1,
  medium: 1.15,
  large: 1.3,
};

/**
 * Returns the configured multiplier for a vehicle size tier.
 */
export function getSizeMultiplier(size: VehicleSize): number {
  return SIZE_MULTIPLIERS[size];
}

/**
 * Returns one adjusted service price using size-based multiplier and whole-dollar rounding.
 */
export function getAdjustedServicePrice(basePrice: number, size: VehicleSize): number {
  return Math.round(basePrice * getSizeMultiplier(size));
}

/**
 * Returns a compact UI label representing size-based price change.
 */
export function formatSizeAdjustmentLabel(size: VehicleSize): string {
  if (size === 'small') {
    return 'Base price';
  }

  const increase = Math.round((getSizeMultiplier(size) - 1) * 100);
  return `+${increase}% by size`;
}
